[CI] MixedClusterClientYamlTestSuiteIT test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/kNN search only} failing

**Build Scans:**
- [elasticsearch-intake #42604 / 9.3.8_bwc-snapshots](https://gradle-enterprise.elastic.co/s/lksce6r2bbuik)
- [elasticsearch-pull-request #159916 / 9.3.8_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/oo3sckwkvxru6)
- [elasticsearch-pull-request #159847 / 9.3.8_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/jp3itgdf6v6h2)
- [elasticsearch-pull-request #159787 / 9.3.8_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ykbl76hsbcqqw)
- [elasticsearch-pull-request #159787 / 9.3.8_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/m43fi5u5ysoik)
- [elasticsearch-pull-request #159746 / 9.3.8_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/xp7gg352awoma)
- [elasticsearch-pull-request #159720 / 9.3.8_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/ouozkwnnbgkse)
- [elasticsearch-pull-request #159708 / 9.3.8_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/zvxljtjzjvs7w)
- [elasticsearch-pull-request #159708 / 9.3.8_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/jtp35a3l6667s)
- [elasticsearch-periodic #12464 / openjdk26_9.3.7_java-matrix-bwc](https://gradle-enterprise.elastic.co/s/xut77wy27riak)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v9.3.8#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/kNN search only}" -Dtests.seed=F86E92047ED7A119 -Dtests.bwc=true -Dtests.locale=dyo-SN -Dtests.timezone=Asia/Seoul -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Dsearch.vectors%2F41_knn_search_byte_quantized_bfloat16%2FkNN%20search%20only%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: org.junit.internal.runners.model.MultipleFailureException: There were 4 errors:
  java.lang.AssertionError(field [hits.hits.0._id] does not match the expected value
Expected: "2"
     but: was "3")
  java.lang.AssertionError(field [hits.hits.0.fields.name.0] does not match the expected value
Expected: "moose.jpg"
     but: was "rabbit.jpg")
  java.lang.AssertionError(field [hits.hits.1._id] does not match the expected value
Expected: "3"
     but: was "2")
  java.lang.AssertionError(field [hits.hits.1.fields.name.0] does not match the expected value
Expected: "rabbit.jpg"
     but: was "moose.jpg")
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/kNN search only}
- [main] 1 consecutive failures in step 9.3.8_bwc-snapshots
- [main] 1 consecutive failures in step openjdk26_9.3.7_java-matrix-bwc
- [main] 1 consecutive failures in step openjdk25_9.3.7_java-matrix-bwc
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 16 failures in test test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/kNN search only} (2.1% fail rate in 774 executions)
- [main] 13 failures in step 9.3.8_part_6_bwc-snapshots (3.9% fail rate in 333 executions)
- [main] 11 failures in pipeline elasticsearch-pull-request (3.0% fail rate in 372 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -619,12 +619,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.commits.StatelessCommitServiceTests
   method: testUnreferencedCommitsAreReleasedAfterRecovery
   issue: https://github.com/elastic/elasticsearch/issues/152730
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/kNN search only}
-  issue: https://github.com/elastic/elasticsearch/issues/152784
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=search.vectors/41_knn_search_byte_quantized_bfloat16/kNN search plus query}
-  issue: https://github.com/elastic/elasticsearch/issues/152786
 - class: org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT
   method: testCloseWhileShardsAreHollowed
   issue: https://github.com/elastic/elasticsearch/issues/151861

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/41_knn_search_byte_quantized_bfloat16.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/41_knn_search_byte_quantized_bfloat16.yml
@@ -78,6 +78,9 @@ setup:
       indices.refresh: { }
 ---
 "kNN search only":
+  - requires:
+      cluster_features: [ "lucene_10_4_upgrade" ]
+      reason: "Approximate int8_hnsw scoring differs between Lucene 10.3 and 10.4+ quantizers"
   - do:
       search:
         index: hnsw_byte_quantized
@@ -112,6 +115,9 @@ setup:
   - match: {hits.hits.1.fields.name.0: "moose.jpg"}
 ---
 "kNN search plus query":
+  - requires:
+      cluster_features: [ "lucene_10_4_upgrade" ]
+      reason: "Approximate int8_hnsw scoring differs between Lucene 10.3 and 10.4+ quantizers"
   - do:
       search:
         index: hnsw_byte_quantized

